File: BinarySearch.cpp
#include<iostream>
#include<vector>
using namespace std;

int ar[10] = {10,20,26,45,55,73,99,102};

int BinarySearch(int ar[],int low,int high,int val){
    int mid = (low+high)/2;
    if(ar[mid] == val){
        return mid;
    } else if(val > ar[mid]){
        BinarySearch(ar,mid+1,high,val);
    } else if(val < ar[mid]){
        BinarySearch(ar,low,mid-1,val);
    } else
        return 0;
}

int main(){
    int num = 45;
    int index = BinarySearch(ar,0,10,num);
    cout<<"Element found at "<<(index+1)<<" position";
    return 0;
}

File: Python/BinarySearch.py
import BubbleSort


def BinarySearch(arr, low, high, x):
    if high >= low:
 
        mid = (high + low) // 2
 
        # If element is present at the middle itself
        if arr[mid] == x:
            return mid
 
        # If element is smaller than mid, then it can only
        # be present in left subarray
        elif arr[mid] > x:
            return BinarySearch(arr, low, mid - 1, x)
 
        # Else the element can only be present in right subarray
        else:
            return BinarySearch(arr, mid + 1, high, x)
 
    else:
        # Element is not present in the array
        return -1


def main():
    ar = [12, 77, 98, 45, 8, 5, 44, 36, 15, 50]
    ar = BubbleSort.BubbleSort(ar)

    #pos = BinarySearch(ar, 0,len(ar)-1, 8)
    print(ar)


if __name__ == '__main__':
    main()
